msg/ZFS-8000-ER/index.html

```diff
@@ -135,13 +135,98 @@
 
 <pre>
 # zpool get freeing
-NAME   PROPERTY  VALUE    SOURCE
-rpool  freeing   0        default
+NAME  PROPERTY  VALUE    SOURCE
+test  freeing   0        default
 </pre>
 
 The ZFS packages may be now be updated and the pool imported.  The on-disk
 format incompatibility can now be corrected online as described in
 <a href="#1">Errata #1</a>.
 </pre>
 </td></tr></table>
+<br><br>
+<hr>
+<br><br>
+<!-- end: templates/success -->
+<!-- start: article -->
+<!-- code: ZFS-8000-ER -->
+<!-- keys: errata -->
+<table width="80%"><tr><td>
+<a name="2"><b>ZFS Errata #3</b></a>
+<p>
+<dl>
+<dt><p><b>Type</b>
+<dd>Compatibility
+<dt><p><b>Severity</b>
+<dd>Moderate
+<dt><p><b>Description</b>
+<dd>An encrypted dataset contains an on-disk format incompatability.
+<dt><p><b>Automated Response</b>
+<dd>No automated response will be taken.
+<dt><p><b>Impact</b>
+<dd>
+Encrypted datasets created before the ZFS packages were updated cannot be
+mounted or opened for write. The errata impacts the ability of ZFS to correctly
+perform raw sends, so these have been disabled for these datasets.
+</dd>
+<dt><p><b>Suggested Action for System Administrator</b>
+<dd>
+System administrators with affected pools will need to recreate any encrypted
+datasets created before the new version of ZFS was used. This can be
+accomplished by using <b>zfs send</b> and <b>zfs receive</b>. Note, however,
+that backups can NOT be done with a raw <b>zfs send -w</b>, since this would
+preserve the on-disk incompatibility. Since the <b>-p</b> and <b>-R</b> options
+imply <b>-w</b> for encrypted datasets, these flags can not be used either.
+Alternatively, system administrators can use conventional tools to back up data
+to new encrypted datasets. The new version of ZFS will prevent new data from
+being written to the impacted datasets, but they can still be mounted read-only.
+<pre>
+# zpool status
+  pool: test
+    id: 1165955789558693437
+ state: ONLINE
+status: Errata #3 detected.
+action: To correct the issue revert to an earlier version and backup
+	and destroy any existing encrypted datasets before updating.
```

Comment:
The action described here doesn't jibe with the recovery procedure outlined below.  There is no need to rollback to a previous version of software unless you need to continue to mount the filesystems read/write.

Reply:
Sorry. That's old. I'll update it